### PR TITLE
Table view on HomePageV2

### PR DESCRIPTION
**Pull request by lihter**
- Added table view to `HomePageTwoVC`
- Removed `moviesCollectionView` and `subcategoryView` from `HomePageV2` (they were only there for testing/demonstration purposes)
- Created `CategoryViewModel` that is very likely to change in the next few PRs (subcategories will be something like let subcategories:` [SubcategoryViewModel]`)
- Created custom `CategoryCell` (for table view) which contains category title `UILabel`, subcategories scroll view `SubcategoryView` and movie collection `UICollectionView`
- Fixed right inset on scrollView (part of `SubcategoryView`)
- Removed placeholder buttons from `SubcategoryView` and replaced them so they get their name from array of `Strings` which they get through populate method (in the future PRs this will change to array of `SubcategoryViewModels`)
- Removed `contentContainer` from custom TV and CV cells and replaced it with _out of the box_ `contentView`
- Added `DesignProtocol` to `FavouriteButton`

### CODE REVIEW

File: MovieApp/MovieApp/App/Features/MovieBrowser/Components/TableViewCategoryCell/CategoryCell.swift [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    }
+    
+    func collectionView(
+        _ collectionView: UICollectionView,
+        cellForItemAt indexPath: IndexPath
+    ) -> UICollectionViewCell {
+        guard let cell = collectionView.dequeueReusableCell(
+                withReuseIdentifier: NewMovieCell.reuseIdentifier,
+                for: indexPath) as? NewMovieCell,
+              let movie = movies?[indexPath.item]
```
Comment (marinzg) [inline MovieApp/MovieApp/App/Features/MovieBrowser/Components/TableViewCategoryCell/CategoryCell.swift:57]: Format this properly.

File: MovieApp/MovieApp/App/Features/MovieBrowser/MoviesViewController/HomePage2/HomePageTwoViewController.swift [hunk 2]
```diff
@@ -3,60 +3,54 @@ import UIKit
...
-        _ collectionView: UICollectionView,
-        cellForItemAt indexPath: IndexPath
-    ) -> UICollectionViewCell {
-        guard let cell = collectionView.dequeueReusableCell(
-                withReuseIdentifier: NewMovieCell.reuseIdentifier,
-                for: indexPath) as? NewMovieCell
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        guard let cell = tableView.dequeueReusableCell(
+                withIdentifier: CategoryCell.reuseIdentifier,
+                for: indexPath) as? CategoryCell
```
Comment (marinzg) [inline MovieApp/MovieApp/App/Features/MovieBrowser/MoviesViewController/HomePage2/HomePageTwoViewController.swift]: Format.

**Review by marinzg — CHANGES_REQUESTED** [review]

File: MovieApp/MovieApp/App/Features/MovieBrowser/Components/TableViewCategoryCell/CategoryCell.swift (hunk 1 above)
Comment (lihter) [reply MovieApp/MovieApp/App/Features/MovieBrowser/Components/TableViewCategoryCell/CategoryCell.swift:57]: [Formatted](https://github.com/lihter/Movie-App/pull/24/files#diff-52168674613714aab4eec989657be9ccf04041400bb872afb308419ec88f2acbR53).

File: MovieApp/MovieApp/App/Features/MovieBrowser/MoviesViewController/HomePage2/HomePageTwoViewController.swift (hunk 2 above)
Comment (lihter) [reply MovieApp/MovieApp/App/Features/MovieBrowser/MoviesViewController/HomePage2/HomePageTwoViewController.swift]: [Fomatted](https://github.com/lihter/Movie-App/pull/24/files#diff-fef596232f7e24aeddc6270887d463ecb9988f34c771cbc593efb85b7b6280e9R41).

**Review by marinzg — APPROVED** [review]

**Pull request merged**